### ISSUE TITLE
trocr fine-tuning on SROIE, is there something wrong with 'gpt2.dict.txt'

### ISSUE THREAD

**Issue by shaoming20798** [microsoft/unilm#502]
Dear author,
    I make a fine-tune on SROIE using trocr method following the codes giving in 'README.md'. And I  evaluate on SROIE using the trained model. But I find many '<\<unk\>>' tokens in the results. Then, I debug the codes, find that the following codes in data.py, line 150,
`python
input_ids = self.target.encode_line(encoded_str, add_if_not_exist=False)
`
many encoded_str has been mapped to label 3, which stands for a unknown token('\<unk\>'), for examples, if the input string is 'TAN WOON YANN', then I will get input_ids equals to 'tensor([3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, ,3, 2], dtype=torch.int32)'. I think this happens because there are some keys missing in the 'gpt2.dict.txt'.
Do I make some mistake? Could you help me to check this out?

**Comment by Dod-o** [comment]
We need to use the bpe parser to get the token id firstly, and then use target_dict to map the id to another vocab, which same as the Roberta in the Fairseq. 

The whole process is:     
text:"TAN WOON YANN"  --（bpe parser）-->  encoded_str:"51 1565 370 46 1340 575 22846"  --(target_dict)-->  input_ids:[565,1889,305,673,2191,854,15118,2]

So it is strange that your encoded_str is text("TAN WOON YANN"). Could you add 2 lines of code in the "SROIETextRecognitionDataset" as below and share me the value of encoded_str and input_ids? I guess maybe the bpe parser is not working?

```
    def __init__(self, root_dir, tfm, bpe_parser, target_dict, crop_img_output_dir=None):
        self.root_dir = root_dir
        self.tfm = tfm            
        self.target_dict = target_dict
        # self.bpe_parser = bpe_parser
        self.ori_data, self.data = SROIETask2(root_dir, bpe_parser, target_dict, crop_img_output_dir)

        # insert these 2 lines.
        encoded_str = bpe_parser.encode('TAN WOON YANN')
        input_ids = self.target_dict.encode_line(encoded_str, add_if_not_exist=False)
```

**Comment by shaoming20798** [comment]
Thanks @Dod-o
Sorry for that I didn't describe my problem clearly. The 'encoder_str' has been actually parsed by bpe parser in the following codes which appear in task.py, line 80:
`python
encoded_str = bpe.encode(content)
`
But I didn't get a encoded_str as in your comments, which is  encoded_str:"51 1565 370 46 1340 575 22846". I got a encoded_str like this:
content:"TAN WOON YANN", encoded_str:"T A N Ġ W O O N Ġ Y A N N". And the corresponding 'input_ids' is 'tensor([3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, ,3, 2], dtype=torch.int32).  
The file 'encoder.json' is downloaded from 'https://dl.fbaipublicfiles.com/fairseq/gpt2_bpe/encoder.json' and the file 'vocab.bpe' is downloaded from 'https://dl.fbaipublicfiles.com/fairseq/gpt2_bpe/vocab.bpe'.
Is there something wrong?

**Comment by shaoming20798** [comment]
> We need to use the bpe parser to get the token id firstly, and then use target_dict to map the id to another vocab, which same as the Roberta in the Fairseq.
> 
> The whole process is: text:"TAN WOON YANN" --（bpe parser）--> encoded_str:"[51 1565 370 46](tel:51 1565 370 46) [1340 575 22846](tel:1340 575 22846)" --(target_dict)--> input_ids:[565,1889,305,673,2191,854,15118,2]
> 
> So it is strange that your encoded_str is text("TAN WOON YANN"). Could you add 2 lines of code in the "SROIETextRecognitionDataset" same as below and share with me the value of encoded_str and input_ids? I guess maybe the bpe parser is not working.
> 
> ```
>     def __init__(self, root_dir, tfm, bpe_parser, target_dict, crop_img_output_dir=None):
>         self.root_dir = root_dir
>         self.tfm = tfm            
>         self.target_dict = target_dict
>         # self.bpe_parser = bpe_parser
>         self.ori_data, self.data = SROIETask2(root_dir, bpe_parser, target_dict, crop_img_output_dir)
> 
>         # insert these 2 lines.
>         encoded_str = bpe_parser.encode('TAN WOON YANN')
>         input_ids = self.target_dict.encode_line(encoded_str, add_if_not_exist=False)
> ```

Thanks @Dod-o 
Sorry for that I didn't describe my problem clearly. The 'encoder_str' has been actually parsed by bpe parser in the following codes which appear in task.py, line 80:
encoded_str = bpe.encode(content)
But I didn't get a encoded_str as in your comments, which is encoded_str:"51 1565 370 46 1340 575 22846". I got a encoded_str like this:
content:"TAN WOON YANN", encoded_str:"T A N Ġ W O O N Ġ Y A N N". And the corresponding 'input_ids' is 'tensor([3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, ,3, 2], dtype=torch.int32).
The file 'encoder.json' is downloaded from 'https://dl.fbaipublicfiles.com/fairseq/gpt2_bpe/encoder.json' and the file 'vocab.bpe' is downloaded from 'https://dl.fbaipublicfiles.com/fairseq/gpt2_bpe/vocab.bpe'.
Is there something wrong?

**Comment by shaoming20798** [comment]
> We need to use the bpe parser to get the token id firstly, and then use target_dict to map the id to another vocab, which same as the Roberta in the Fairseq.
> 
> The whole process is: text:"TAN WOON YANN" --（bpe parser）--> encoded_str:"51 1565 370 46 1340 575 22846" --(target_dict)--> input_ids:[565,1889,305,673,2191,854,15118,2]
> 
> So it is strange that your encoded_str is text("TAN WOON YANN"). Could you add 2 lines of code in the "SROIETextRecognitionDataset" same as below and share with me the value of encoded_str and input_ids? I guess maybe the bpe parser is not working.
> 
> ```
>     def __init__(self, root_dir, tfm, bpe_parser, target_dict, crop_img_output_dir=None):
>         self.root_dir = root_dir
>         self.tfm = tfm            
>         self.target_dict = target_dict
>         # self.bpe_parser = bpe_parser
>         self.ori_data, self.data = SROIETask2(root_dir, bpe_parser, target_dict, crop_img_output_dir)
> 
>         # insert these 2 lines.
>         encoded_str = bpe_parser.encode('TAN WOON YANN')
>         input_ids = self.target_dict.encode_line(encoded_str, add_if_not_exist=False)
> ```

Maybe I have find the problem. I comment the bpe encoder codes. I will try it again and see if it works.
